### PR TITLE
fix: added debounce on ds creation name input

### LINKED ISSUE #364: Sluggish text entry when naming a dataset on deploy
image

I believe this field is validating the name on each keystroke.  When typing quickly, there is a noticeable delay that prevents the text from showing up immediately.

Investigate and see if we can debounce or otherwise make typing into this field happen more smoothly.

### PR DESCRIPTION
**Pull request by boandriy**
added debounce on ds creation name input, so after user stops typing - check for errors

fixes #364

### CODE REVIEW

File: src/features/workflow/modal/DeployModal.tsx [hunk 1]
```diff
@@ -47,8 +48,10 @@ const DeployModal: React.FC = () => {
 
   // local state
   const [ dsName, setDsName ] = useState('')
-  const [ dsNameError, setDsNameError ] = useState()
+  const [debouncedDsName] = useDebounce(dsName, 500)
```
Comment (b5) [inline src/features/workflow/modal/DeployModal.tsx]: whenever you're using a debounce, declare the interval at a file-level constant just after the block of import statements:

```ts
const debounce_rate = 500
```

That way if we need to tune it later it's visible & clear

File: src/features/workflow/modal/DeployModal.tsx [hunk 2]
```diff
@@ -85,12 +88,14 @@ const DeployModal: React.FC = () => {
...
   const handleDsNameChange = (value: string) => {
-    //validate the dataset name
-    const err = validateDatasetName(value)
-    setDsNameError(err)
     setDsName(value)
   }
 
+  useEffect(() => {
+    if (isNew)
+      setCanBeDeployed(false)
```
Comment (b5) [inline src/features/workflow/modal/DeployModal.tsx]: always use curly braces on `if` statements

File: src/features/workflow/modal/DeployModal.tsx [hunk 3]
```diff
@@ -103,18 +108,14 @@ const DeployModal: React.FC = () => {
...
-  const checkReadyToDeploy = () => {
-
+  useEffect(() => {
     if (isNew) {
-      // if the name exists and is valid
-      if (!dsName.length || dsNameError) { return false }
+      //validate the dataset name
+      const err = validateDatasetName(debouncedDsName)
+      setDsNameError(err)
+      setCanBeDeployed(!(!debouncedDsName.length || !!err))
```
Comment (b5) [inline src/features/workflow/modal/DeployModal.tsx]:
```ts
setCanBeDeployed(debouncedDsName.length > 0 && err === null)
```

File: src/features/workflow/modal/DeployModal.tsx (hunk 2 above)
Comment (boandriy) [reply src/features/workflow/modal/DeployModal.tsx]: Noted

**Review by b5 — APPROVED** [review]
LGTM

**Pull request merged**